content/engineering/guest-post/introduction-to-solidjs/index.md

```diff
@@ -0,0 +1,766 @@
+
+SolidJS allows you to create reactive variables called *signals* using the `createSignal` function. This function takes in the initial value of the variable as a parameter. And it returns an array where the first element is the variable itself and the second element is a function that is used to mutate that variable.
+
+[Here's an example](https://playground.solidjs.com/?hash=776620219&version=1.4.1) that changes the reactive variable called `name`:
+
+```jsx
+import { render } from "solid-js/web";
+import { createSignal } from "solid-js";
+
+function DisplayName() {
+  const [name, setName] = createSignal('fuzzysid');
+  const changeName = () => setName('siddhant')
+
+  return (
+    <>
+      <h1> {name} </h1>
+      <button onClick={changeName}> Change Name </button>
+    </>
+  );
+}
+
+render(() => <DisplayName />, document.getElementById("app")!);
+
+```
+
+Initially, the name outputs the string 'fuzzysid':
+
+![Reactivity Before Update](./reactivity-before-update.png)
+
+A click event ia used on the button that fires a function called `changeName`. When you press the *Change Name* button, the `changeName` function updates the `name` signal to 'siddhant':
+
+![Reactivity After Update](./reactivity-after-update.png)
+
+What a `state` is to React components, a `signal` is to Solid components. [Here's another useful and typical example](https://playground.solidjs.com/?hash=-1579873004&version=1.4.1) of reactive variables:
+
+```jsx
+import { render } from "solid-js/web";
+import { createSignal } from "solid-js";
+
+function Counter() {
+  const [count,setCount] = createSignal(0);
+
+  return (
+    <>
+      <h1> Counter: {count} </h1>
+      <button onClick={()=>setCount(count=>count-1)}>  Decrement </button>
+      &nbsp;
+      <button onClick={()=>setCount(count=>count+1)}>  Increment </button>
+    </>
+  );
+}
+
+render(() => <Counter />, document.getElementById("app")!);
+```
+
+You can take the current value of a signal inside the signal's mutation's callback function. The above code used the current value of `count` to increment and decrement it accordingly. 
+
+### Effects
+
+Effects are triggers or watchers that can help you fire a function when a signal updates.
+
+Let's say in the previous example you want to throw an alert when count becomes greater than 5.
+
+You can create an effect using the `createEffect` function to do so. It takes a callback function as a parameter. This callback function is fired whenever a signal invoked inside it changes. 
+
+```jsx
+createEffect(()=>{
+    if(count()>5) alert('Count is greater than five')
+})
+```
+
+So now if you click on the *Increment* button, you should get an alert when it becomes greater than 5:
+
+![Effects Example](./effects-example.png)
+
+### Conditionals
+
+Solid provides special components called `<Show>` to render a template based on a conditioner the value of a signal. [Here's an example of that](https://playground.solidjs.com/?hash=1279589651&version=1.4.1):
```

Comment:
Hi @FuzzySid The following part in this sentence is a bit unclear. Please clarify:

`to render a template based on a conditioner the value of a signal.`

Reply:
I have rephrased this line for more clarity.